src/lib.rs

```diff
@@ -215,6 +215,39 @@ impl<T: Hash, S: BuildHasher> HashRing<T, S> {
     }
 }
 
+pub struct HashRingIterator<T> {
+    ring: std::vec::IntoIter<T>,
+    index: usize,
```

Comment:
Can we get rid of `index` now that we're wrapping the underlying vector iterator?

Reply:
good call